src/Illuminate/Database/Query/Builder.php

```diff
@@ -1593,7 +1604,7 @@ public function whereJsonLength($column, $operator, $value = null, $boolean = 'a
         $this->wheres[] = compact('type', 'column', 'operator', 'value', 'boolean');
 
         if (! $value instanceof Expression) {
-            $this->addBinding((int) $value);
+            $this->addBinding((int) $this->scalarValue($value));
```

Comment:
Is this necessary as we are casting to an integer?

Reply:
I can’t really say. Casting an array into integer returns 1 - so there’s that. Really depends on what’s expected when one passes an array there.